diff --git a/builds.toml b/builds.toml
index 8e0a131..7b4cfde 100644
--- a/builds.toml
+++ b/builds.toml
@@ -22,6 +22,7 @@ builds = [
     "bookworm-rt-am62xxsip-evm",
     "bookworm-am64xx-evm",
     "bookworm-rt-am64xx-evm",
+    "bookworm-j7200-evm",
 ]
 
 # List of all valid trixie builds
@@ -126,3 +127,7 @@ builds = [
     distro_codename = "bookworm"
     rt_linux = "true"
 
+[bookworm-j7200-evm]
+    machine = "j7200-evm"
+    distro_codename = "bookworm"
+    rt_linux = "false"
diff --git a/configs/bdebstrap_configs/bookworm/bookworm-j7200-evm.yaml b/configs/bdebstrap_configs/bookworm/bookworm-j7200-evm.yaml
new file mode 100644
index 0000000..521cb6b
--- /dev/null
+++ b/configs/bdebstrap_configs/bookworm/bookworm-j7200-evm.yaml
@@ -0,0 +1,84 @@
+---
+mmdebstrap:
+  architectures:
+    - arm64
+  mode: auto
+  keyrings:
+    - /usr/share/keyrings/debian-archive-keyring.gpg
+  suite: bookworm
+  variant: standard
+  hostname: j7200-evm
+  components:
+    - main
+    - contrib
+    - non-free-firmware
+  packages:
+    - build-essential
+    - gpg
+    - curl
+    - firmware-ti-connectivity
+    - init
+    - iproute2
+    - less
+    - libdrm-dev
+    - libpam-systemd
+    - locales
+    - neofetch
+    - network-manager
+    - net-tools
+    - openssh-server
+    - sudo
+    - vim
+    - k3conf
+    - weston
+    - linux-image-6.1.83-k3
+    - linux-headers-6.1.83-k3
+    - linux-libc-dev
+    - cryptodev-linux-dkms
+    - firmware-ti-ipc-j7200
+    - firmware-cnm-wave
+    - firmware-ti-eth-j7200
+    - libti-rpmsg-char
+    - libti-rpmsg-char-dev
+    - libpru-pssp-dev
+    - pru-pssp
+    - parted
+    - e2fsprogs
+  mirrors:
+    - http://deb.debian.org/debian
+  setup-hooks:
+      # Setup TI Debian Package Repository
+    - 'mkdir -p $1/etc/apt/sources.list.d/'
+    - 'wget https://raw.githubusercontent.com/TexasInstruments/ti-debpkgs/main/ti-debpkgs.sources -P $1/etc/apt/sources.list.d/'
+      # Setup Apt repository preferences
+    - 'mkdir -p $1/etc/apt/preferences.d/'
+    - 'printf "Package: *\nPin: origin TexasInstruments.github.io\nPin-Priority: 1001" >> $1/etc/apt/preferences.d/ti-debpkgs'
+      # Setup Kernel post-install scripts
+    - 'mkdir -p $1/etc/kernel/postinst.d/'
+    - 'echo "PWD = $PWD"'
+    - 'upload target/kernel/postinst.d/cp-kernel-and-overlays /etc/kernel/postinst.d/cp-kernel-and-overlays'
+    - 'chmod a+x $1/etc/kernel/postinst.d/cp-kernel-and-overlays'
+  essential-hooks:
+    # FIXME: Find a better workaround instead of sleep
+    - 'sleep 10' # workaround for /proc resource busy unable to umount issue
+  customize-hooks:
+      # Remove passwd for root user
+    - 'chroot "$1" passwd --delete root'
+      # Fix apt install mandb permission issue
+    - 'chroot "$1" chown -R man: /var/cache/man/'
+    - 'chroot "$1" chmod -R 755 /var/cache/man/'
+      # update packages to avoid mandatory update after first boot
+    - 'chroot "$1" apt-get update'
+      # Setup .bashrc for clean command-line experience
+    - 'chroot "$1" cp /etc/skel/.bashrc ~/.bashrc'
+      # Enable ssh to root user without password
+    - 'chroot "$1" echo "PermitRootLogin yes" >> $1/etc/ssh/sshd_config'
+    - 'chroot "$1" echo "PermitEmptyPasswords yes" >> $1/etc/ssh/sshd_config'
+      # Resize Rootfs Service
+    - 'chroot "$1" mkdir -p /usr/bin'
+    - 'upload target/resize_rootfs/resize_rootfs.sh /usr/bin/resize_rootfs.sh'
+    - 'chroot "$1" chmod a+x /usr/bin/resize_rootfs.sh'
+    - 'chroot "$1" mkdir -p /etc/systemd/system/'
+    - 'upload target/resize_rootfs/resize_rootfs.service /etc/systemd/system/resize_rootfs.service'
+    - '$BDEBSTRAP_HOOKS/enable-units "$1" resize_rootfs'
+
diff --git a/configs/bsp_sources.toml b/configs/bsp_sources.toml
index 988f2f1..a73b429 100644
--- a/configs/bsp_sources.toml
+++ b/configs/bsp_sources.toml
@@ -13,8 +13,8 @@
     linux_fw_srcrev = "10.01.10"
 
 [09.02.00.010]
-    atf_srcrev = "00f1ec6b8740ccd403e641131e294aabacf2a48b"
-    optee_srcrev = "012cdca49db398693903e05c42a254a3a0c0d8f2"
-    uboot_srcrev = "09.02.00.010"
-    linux_fw_srcrev = "09.02.00.010"
+    atf_srcrev = "58b25570c9ef91753b14c2103f45f4be9dddb696"
+    optee_srcrev = "4.4.0"
+    uboot_srcrev = "10.01.08"
+    linux_fw_srcrev = "ti-linux-firmware"
 
diff --git a/configs/machines/09.02.00.010.toml b/configs/machines/09.02.00.010.toml
index d6e44b0..ca107b1 100644
--- a/configs/machines/09.02.00.010.toml
+++ b/configs/machines/09.02.00.010.toml
@@ -44,3 +44,12 @@
     optee_make_args = "."
     uboot_r5_defconfig = "am64x_evm_r5_defconfig"
     uboot_a53_defconfig = "am64x_evm_a53_defconfig"
+
+[j7200-evm]
+    # u-boot config
+    atf_target_board = "generic"
+    atf_make_args="K3_PM_SYSTEM_SUSPEND=1"
+    optee_platform = "k3-j721e"
+    optee_make_args = "ta-targets=ta_arm64"
+    uboot_r5_defconfig = "j7200_evm_r5_defconfig"
+    uboot_a72_defconfig = "j7200_evm_a72_defconfig"
diff --git a/patches/ti-u-boot/0001-tiboot3-default-dtb-fix.patch b/patches/ti-u-boot/0001-tiboot3-default-dtb-fix.patch
new file mode 100644
index 0000000..37f054c
--- /dev/null
+++ b/patches/ti-u-boot/0001-tiboot3-default-dtb-fix.patch
@@ -0,0 +1,34 @@
+diff --git a/arch/arm/dts/k3-j7200-binman.dtsi b/arch/arm/dts/k3-j7200-binman.dtsi
+index a524b246187..52d83be9425 100644
+--- a/arch/arm/dts/k3-j7200-binman.dtsi
++++ b/arch/arm/dts/k3-j7200-binman.dtsi
+@@ -188,6 +188,7 @@
+ &binman {
+ 	tiboot3-j7200_sr2-hs-fs-evm.bin {
+ 		filename = "tiboot3-j7200_sr2-hs-fs-evm.bin";
++		symlink = "tiboot3.bin";
+ 		ti-secure-rom {
+ 			content = <&u_boot_spl_fs>, <&ti_fs_enc_fs>, <&combined_tifs_cfg_fs>,
+ 				<&combined_dm_cfg_fs>, <&sysfw_inner_cert_fs>;
+@@ -233,7 +234,7 @@
+ &binman {
+ 	tiboot3-j7200-gp-evm.bin {
+ 		filename = "tiboot3-j7200-gp-evm.bin";
+-		symlink = "tiboot3.bin";
++		/* symlink = "tiboot3.bin"; */
+ 		ti-secure-rom {
+ 			content = <&u_boot_spl_unsigned>, <&ti_fs_gp>,
+ 				<&combined_tifs_cfg_gp>, <&combined_dm_cfg_gp>;
+diff --git a/board/ti/j721e/evm.c b/board/ti/j721e/evm.c
+index 90d3411e2ff..c6103624b16 100644
+--- a/board/ti/j721e/evm.c
++++ b/board/ti/j721e/evm.c
+@@ -401,7 +401,7 @@ void configure_serdes_sierra(void)
+ static struct ti_fdt_map ti_j721e_evm_fdt_map[] = {
+ 	{"j721e", "k3-j721e-common-proc-board.dtb"},
+ 	{"j721e-sk", "k3-j721e-sk.dtb"},
+-	{"j7200", "k3-j7200-common-proc-board.dtb"},
++	{"j7200", "k3-j7200-evm.dtb"},
+ 	{ /* Sentinel. */ }
+ };
+ static void setup_board_eeprom_env(void)
diff --git a/scripts/build_bsp.sh b/scripts/build_bsp.sh
index 2b7b2b9..d877a11 100755
--- a/scripts/build_bsp.sh
+++ b/scripts/build_bsp.sh
@@ -131,7 +131,8 @@ bsp_version=$2
 
     uboot_r5_defconfig=($(read_machine_config ${machine} uboot_r5_defconfig ${bsp_version}))
     uboot_r5_defconfig=`echo $uboot_r5_defconfig | tr ',' ' '`
-    uboot_a53_defconfig=($(read_machine_config ${machine} uboot_a53_defconfig ${bsp_version}))
+    uboot_a72_defconfig=($(read_machine_config ${machine} uboot_a72_defconfig ${bsp_version}))
+    platform=($(read_machine_config ${machine} atf_target_board ${bsp_version}))
 
     cd ${UBOOT_DIR}
     log "> uboot-r5: building .."
@@ -140,11 +141,11 @@ bsp_version=$2
     cp ${UBOOT_DIR}/out/r5/tiboot3*.bin ${topdir}/build/${build}/tisdk-debian-${distro}-boot/ &>> ${LOG_FILE}
 
     cd ${UBOOT_DIR}
-    log "> uboot-a53: building .."
-    make -j`nproc` ARCH=arm CROSS_COMPILE=${cross_compile} ${uboot_a53_defconfig} O=${UBOOT_DIR}/out/a53 &>>"${LOG_FILE}"
-    make -j`nproc` ARCH=arm CROSS_COMPILE=${cross_compile} BL31=${TFA_DIR}/build/k3/lite/release/bl31.bin TEE=${OPTEE_DIR}/out/arm-plat-k3/core/tee-pager_v2.bin O=${UBOOT_DIR}/out/a53 BINMAN_INDIRS=${topdir}/build/${build}/bsp_sources/ti-linux-firmware &>>"${LOG_FILE}"
-    cp ${UBOOT_DIR}/out/a53/tispl.bin ${topdir}/build/${build}/tisdk-debian-${distro}-boot/ &>> ${LOG_FILE}
-    cp ${UBOOT_DIR}/out/a53/u-boot.img ${topdir}/build/${build}/tisdk-debian-${distro}-boot/ &>> ${LOG_FILE}
+    log "> uboot-a72: building .."
+    make -j`nproc` ARCH=arm CROSS_COMPILE=${cross_compile} ${uboot_a72_defconfig} O=${UBOOT_DIR}/out/a72 &>>"${LOG_FILE}"
+    make -j`nproc` ARCH=arm CROSS_COMPILE=${cross_compile} BL31=${TFA_DIR}/build/k3/${platform}/release/bl31.bin TEE=${OPTEE_DIR}/out/arm-plat-k3/core/tee-pager_v2.bin BINMAN_INDIRS=${FW_DIR} O=${UBOOT_DIR}/out/a72 &>>"${LOG_FILE}"
+    cp ${UBOOT_DIR}/out/a72/tispl.bin* ${topdir}/build/${build}/tisdk-debian-${distro}-boot/ &>> ${LOG_FILE}
+    cp ${UBOOT_DIR}/out/a72/u-boot.img* ${topdir}/build/${build}/tisdk-debian-${distro}-boot/ &>> ${LOG_FILE}
 
 	case ${machine} in
 		am62pxx-evm | am62xx-evm | am62xx-lp-evm | am62xxsip-evm)
